src/m365/spo/commands/site/site-admin-add.ts

```diff
@@ -0,0 +1,358 @@
+      await this.setPrimaryAdminAsAdmin(adminUrl, siteId, loginNameToAdd);
+    }
+  }
+
+  private async getSiteIdBasedOnUrl(siteUrl: string, logger: Logger): Promise<string> {
+    const siteGraphId = await spo.getSiteId(siteUrl, logger, this.verbose);
+    const match = siteGraphId.match(/,([a-f0-9\-]{36}),/i);
+    if (!match) {
+      throw `Site with URL ${siteUrl} not found`;
+    }
+
+    return match[1];
+  }
+
+  private async getSiteAdmins(adminUrl: string, siteId: string): Promise<AdminUserResult[]> {
+    const requestOptions: CliRequestOptions = {
+      url: `${adminUrl}/_api/SPO.Tenant/GetSiteAdministrators?siteId='${siteId}'`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        'content-type': 'application/json;charset=utf-8'
+      }
+    };
+
+    const response: string = await request.post<string>(requestOptions);
+    const responseContent: AdminResult = JSON.parse(response);
+    return responseContent.value;
+  }
+
+  private async getCorrectLoginName(options: Options): Promise<string> {
+    if (options.userId || options.userName) {
+      let requestUrl: string = `https://graph.microsoft.com/v1.0/users`;
+      if (options.userName) {
+        requestUrl += `('${formatting.encodeQueryParameter(options.userName)}')`;
+      }
+      else {
+        requestUrl += `/${options.userId}`;
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: requestUrl,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      const user = await request.get<IGraphUser>(requestOptions);
+      if (user && user.userPrincipalName) {
+        return `i:0#.f|membership|${user.userPrincipalName}`;
+      }
+
+      throw 'User not found.';
+    }
+    else {
+      const group = options.groupId ? await entraGroup.getGroupById(options.groupId) : await entraGroup.getGroupByDisplayName(options.groupName!);
+      //for entra groups, M365 groups have an associated email and security groups don't
+      if (group?.mail) {
+        //M365 group is prefixed with c:0o.c|federateddirectoryclaimprovider
+        return `c:0o.c|federateddirectoryclaimprovider|${group.id}`;
+      }
+      else {
+        //security group is prefixed with c:0t.c|tenant
+        return `c:0t.c|tenant|${group?.id}`;
+      }
+    }
+  }
+
+  private async setSiteAdminsAsAdmin(adminUrl: string, siteId: string, admins: string[]): Promise<void> {
+    const requestOptions: CliRequestOptions = {
+      url: `${adminUrl}/_api/SPOInternalUseOnly.Tenant/SetSiteSecondaryAdministrators`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        'content-type': 'application/json;charset=utf-8'
+      },
+      data: {
+        secondaryAdministratorsFieldsData: {
+          siteId: siteId,
+          secondaryAdministratorLoginNames:
+            admins
+        }
+      }
+    };
+
+    await request.post(requestOptions);
```

Comment:
We can just use return request.post() here.

Reply:
done